packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,295 @@
+///      2. The Safe must then configure the module by calling configure() with params
+contract LivenessModule2 is ISemver {
+    /// @notice Configuration for a Safe's liveness module
+    struct ModuleConfig {
+        uint256 livenessResponsePeriod;
+        address fallbackOwner;
+    }
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => ModuleConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to active challenge start time (0 if none)
+    mapping(address => uint256) public challengeStartTime;
+
+    /// @notice Reserved address used as previous owner to the first owner in a Safe
+    address internal constant SENTINEL_OWNER = address(0x1);
+
+    /// @notice Error for when module is not enabled for the Safe
+    error LivenessModule2_ModuleNotEnabled();
+
+    /// @notice Error for when Safe is not configured for this module
+    error LivenessModule2_ModuleNotConfigured();
+
+    /// @notice Error for when a challenge already exists
+    error LivenessModule2_ChallengeAlreadyExists();
+
+    /// @notice Error for when no challenge exists
+    error LivenessModule2_ChallengeDoesNotExist();
+
+    /// @notice Error for when trying to cancel a challenge after response period has ended
+    error LivenessModule2_ResponsePeriodEnded();
+
+    /// @notice Error for when trying to execute ownership transfer while response period is active
+    error LivenessModule2_ResponsePeriodActive();
+
+    /// @notice Error for when caller is not authorized
+    error LivenessModule2_UnauthorizedCaller();
+
+    /// @notice Error for invalid response period
+    error LivenessModule2_InvalidResponsePeriod();
+
+    /// @notice Error for invalid fallback owner
+    error LivenessModule2_InvalidFallbackOwner();
+
+    /// @notice Error for when trying to clear configuration while module is enabled
+    error LivenessModule2_ModuleStillEnabled();
+
+    /// @notice Error for when ownership transfer verification fails
+    error LivenessModule2_OwnershipTransferFailed();
+
+    /// @notice Emitted when a Safe enables the module
+    event ModuleEnabled(address indexed safe, uint256 livenessResponsePeriod, address fallbackOwner);
+
+    /// @notice Emitted when a Safe disables the module
+    event ModuleDisabled(address indexed safe);
+
+    /// @notice Emitted when a challenge is started
+    event ChallengeStarted(address indexed safe, uint256 challengeStartTime);
+
+    /// @notice Emitted when a challenge is cancelled
+    event ChallengeCancelled(address indexed safe);
+
+    /// @notice Emitted when ownership is transferred to the fallback owner
+    event ChallengeSucceeded(address indexed safe, address fallbackOwner);
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Returns challenge_start_time + liveness_response_period if challenge exists, or
+    /// 0 if not
+    /// @dev MUST never revert
```

Comment:
A bit of a nit, the `@dev` comments here create a coupling with the spec - this is a surface that we'd need to maintain and will likely get out of line with the specs. I'd recommend keeping the invariants as part of the specs (and not `@dev` comments) and making sure we maintain the specs carefully.

Reply:
Fixed in e03a5d7dd728851d60cf929827f0cbec5f1803ee